File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        count = [0]*(n+1)
        for i in set(nums):
            x = nums.count(i)
            if count[x] == 0:
                count[x] = [i]
            else:
                count[x].append(i)

        z = 0
        l = []
        for j in count[::-1]:
            if j:
                for num in j:
                    z += 1
                    l.append(num)
            if z == k:
                break

        return l

File: Data Structures & Algorithms/three-integer-sum/submission-1.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> answer;
        sort(nums.begin(), nums.end());
        int n = nums.size();
        for (int i = 0; i < n; i++){
            if (i > 0 && nums[i-1] == nums[i]) continue;
            int l = i + 1;
            int r = n - 1;
            while (l < r){
                int sum = nums[i] + nums[l] + nums[r];
                if (sum == 0){
                    answer.push_back({nums[i], nums[l], nums[r]});
                    l++;
                    while (l < n && nums[l] == nums[l-1]) l++;
                } else if (sum < 0){
                    l++;
                } else r--;
            }

        }
        return answer;
    }
};
